src/licensedcode/detection.py

```diff
@@ -263,6 +370,278 @@ def matched_text(
             for m in self.matches
         )
 
+    def percentage_license_text_of_file(self, qspans):
+        from licensedcode.spans import Span
+
+        matched_tokens_length = len(Span().union(*qspans))
+        query_tokens_length = self.query.tokens_length(with_unknown=True)
+        return round((matched_tokens_length / query_tokens_length) * 100, 2)
+    
+    def to_dict(
+        self, 
+        include_text=False,
+        license_text_diagnostics=False,
+        license_url_template=SCANCODE_LICENSEDB_URL,
+    ):
+        matches = self.matches
+        data_matches = []
+
+        for match in matches:
+            data_matches.extend(
+                _licenses_data_from_match(
+                    match=match,
+                    include_text=include_text,
+                    license_text_diagnostics=license_text_diagnostics,
+                    license_url_template=license_url_template,
+                )
+            )
+
+        detection = attr.asdict(self)
+        detection["matches"] = data_matches
+
+        return detection
+
+
+def _licenses_data_from_match(
+    match,
+    include_text=False,
+    license_text_diagnostics=False,
+    license_url_template=SCANCODE_LICENSEDB_URL,
+):
+    """
+    Return a list of "licenses" scan data built from a license match.
+    Used directly only internally for testing.
```

Comment:
May be this is best in the tests/  then?

Reply:
This was being used in the serialization actually earlier too, still being used. Will update  the docstring.